plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -3754,6 +3763,54 @@ private void testMultipleWriters(Session session, HiveStorageFormat storageForma
         }
     }
 
+    @Test
+    public void testMultipleWritersWhenTaskScaleWritersIsEnabled()
+    {
+        long workers = (long) computeScalar("SELECT count(*) FROM system.runtime.nodes");
+        int minScaledWriterCount = new TaskManagerConfig().getMinScaledWriterCount();
+        testWithAllStorageFormats((session, storageFormat) ->
+                assertFilesCount(session, storageFormat, DataSize.of(1, MEGABYTE))
+                        .isBetween(workers + 1, workers * minScaledWriterCount));
+    }
+
+    @Test
+    public void testSingleWriterWhenTaskScaleWritersIsEnabled()
+    {
+        testWithAllStorageFormats((session, storageFormat) ->
+                assertFilesCount(session, storageFormat, DataSize.of(2, GIGABYTE))
+                        .isEqualTo(1));
+    }
+
+    private AbstractLongAssert<?> assertFilesCount(
+            Session session,
+            HiveStorageFormat storageFormat,
+            DataSize writerMinSize)
+    {
+        String tableName = "task_scale_writers_" + randomTableSuffix();
+        try {
+            @Language("SQL") String createTableSql = format(
+                    "CREATE TABLE %s WITH (format = '%s') AS SELECT * FROM tpch.sf1.orders",
+                    tableName,
+                    storageFormat);
+            assertUpdate(
+                    Session.builder(session)
+                            .setSystemProperty(SCALE_WRITERS, "true")
```

Comment:
It would be better to disable SCALE_WRITERS here to reduce the variables in this test